taichi/transforms/demote_operations.cpp

```diff
@@ -61,25 +61,6 @@ class DemoteOperations : public BasicStmtVisitor {
     return floor;
   }
 
-  void visit(BitExtractStmt *stmt) override {
```

Comment:
I wonder in our definition does BitExtractStmt does sign extension? Does it depend on the input type? If so then we should add a case to detect input types and choose one between `OpBitFieldUExtract` and `OpBitFieldSExtract`

Reply:
Nope. `BitExtractStmt` doesn't perform sign extension.